it/controllers/benefits/medical/MedicalOrDentalBenefitsAmountControllerISpec.scala

```diff
@@ -0,0 +1,583 @@
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(user.commonExpectedResults.expectedCaption(taxYearEOY))
+          buttonCheck(user.commonExpectedResults.continueButtonText, continueButtonSelector)
+          textOnPageCheck(user.specificExpectedResults.get.expectedPreAmountParagraph(insuranceAmount), paragraphTextSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraph, paragraphTextSelector(3))
+          textOnPageCheck(user.specificExpectedResults.get.expectedParagraphforForm, paragraphTextSelector(4))
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck("100", amountField)
+          formPostLinkCheck(continueLink(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+
+        }
+      }
+    }
+
+    "redirect to another page when the request is valid but they aren't allowed to view the page and" should {
+
+      val user = UserScenario(isWelsh = false, isAgent = false, CommonExpectedEN, Some(ExpectedAgentEN))
+
+      "Redirect user to the check your benefits page with no cya data in session" which {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          authoriseAgentOrIndividual(user.isAgent)
+          urlGet(url(taxYearEOY), user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(CheckYourBenefitsController.show(taxYearEOY, employmentId).url)
+        }
+      }
+
+      "Redirect user to the tax overview page when in year" which {
+        lazy val result: WSResponse = {
+          dropEmploymentDB()
+          userDataStub(userData(fullEmploymentsModel(hmrcEmployment = Seq(employmentDetailsAndBenefits(
+            fullBenefits.map(_.copy(benefits = fullBenefits.flatMap(_.benefits.map(_.copy(telephone = None))))))))), nino, taxYear)
+          insertCyaData(employmentUserData(isPrior = false, cyaModel("name", hmrc = true,
+            Some(benefits(fullMedicalChildcareEducationModel)))), userRequest)
+          authoriseAgentOrIndividual(user.isAgent)
+          urlGet(url(taxYear), user.isWelsh, follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYear)))
+        }
+
+        "has an SEE_OTHER(303) status" in {
+          result.status shouldBe SEE_OTHER
+          result.header("location") shouldBe Some(s"http://localhost:11111/income-through-software/return/$taxYear/view")
+        }
+      }
+
+      "redirect to the other services question page when there is a medical insurance amount but the other medicalInsuranceQuestion has not been answered" +
```

Comment:
i think this test name can be renamed as its quite confusing as to what its saying is going on